src/librustc/util/ppaux.rs

```diff
@@ -925,9 +925,11 @@ impl<'tcx> fmt::Display for ty::ProjectionPredicate<'tcx> {
 
 impl<'tcx> fmt::Display for ty::ProjectionTy<'tcx> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        let item_name = ty::tls::with(|tcx| self.item_name(tcx));
+        let (trait_ref, item_name) = ty::tls::with(
+            |tcx| (self.trait_ref(tcx), self.item_name(tcx))
+        );
```

Comment:
Although I don't think I agree with the indentation, the closure arguments should be on the previous line.